packages/http/src/router/__tests__/index.spec.ts

```diff
@@ -42,31 +41,14 @@ describe('http router', () => {
     });
 
     describe('given a resource', () => {
-      test('should not match if no server defined', () => {
+      test('should match even if no server defined', () => {
         const method = pickOneHttpMethod();
         const path = randomPath();
+        const expectedResource = createResource(method, path, []);
 
-        return expect(() =>
+        return expect(
           router.route({
-            resources: [createResource(method, path, [])],
-            input: {
-              method,
-              url: {
-                baseUrl: '',
-                path,
-              },
-            },
-          })
-        ).toThrow(NO_SERVER_CONFIGURATION_PROVIDED_ERROR);
-      });
-
-      test('should not match if no servers arrays provided', () => {
-        const method = pickOneHttpMethod();
-        const path = randomPath();
-
-        return expect(() =>
-          router.route({
-            resources: [createResource(method, path, [])],
+            resources: [expectedResource],
```

Comment:
This test verifies whether router responds with a resource or not in certain conditions. This is not about selecting particular resource out of many.

But you are completely right - it's misleading. I changed the test so it only checks whether there was no error thrown without caring what resource has been returned.

Reply:
> As I mentioned multiple times, tests should fail by default, not pass, otherwise they do not tell us anything.

I'm bit unsure if I understand this correctly. How this particular test should look like?

Reply:
Ok, the new version clarifies what's going on. Resolved!